### PR TITLE
Update tutorial screens

### PR DESCRIPTION
**Pull request by Giancarlo-Castaneda**
# Ticket
[CAR-35](https://team-h33xjxzctfzs.atlassian.net/browse/CAR-35)
# Description
Se refactorizan las pantallas del splash y el tutorial de uso para nuevos usuarios.
TODO
- [ ] agregar la mensajes definidos por producto, (están en revisión)

# Screenshots
|Splash|
|:---:|
||

|Tutorial|
|:---:|
||

### CODE REVIEW

File: CarGurus App/lib/presentation/features/splash/splash_screen.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  const SplashScreen({super.key});
+
+  @override
+  State<SplashScreen> createState() => _SplashScreen();
+}
+
+class _SplashScreen extends State<SplashScreen> {
+  late ColorNotifire notifire;
+
+  getdarkmodepreviousstate() async {
```
Comment (ndmunoz10) [inline CarGurus App/lib/presentation/features/splash/splash_screen.dart:22]: Change method name to camelCase

File: CarGurus App/lib/presentation/features/splash/splash_screen.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+          builder: (context) => const OnbordingScreen(),
+        ),
+      );
+    });
+  }
+
+  getLocationPermition() async {
+    LocationPermission permission;
+    permission = await Geolocator.checkPermission();
+    permission = await Geolocator.requestPermission();
```
Comment (ndmunoz10) [inline CarGurus App/lib/presentation/features/splash/splash_screen.dart:50]: Please re check this flow, we're checking if the permission is granted but still requesting it even if the permission is granted. Also, do we need to request for location permissions?

**Review by ndmunoz10 — CHANGES_REQUESTED** [review]

File: CarGurus App/lib/presentation/features/onboarding/onbording_screen.dart [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+                    image: DecorationImage(
+                      image: AssetImage(_backgroundImageName(index)),
+                      fit: index == 0 ? BoxFit.cover : null,
+                    ),
+                  ),
+                );
+              },
+              onPageChanged: (value) {
+                setState(() {
+                  index = value;
```
[the remark this answers is not in the document]
Comment (Giancarlo-Castaneda) [reply CarGurus App/lib/presentation/features/onboarding/onbording_screen.dart]: I implemented cubit but I still need to have an index in the screen file to be able to change the pagination indicator, do you know if there is a way to have this property within cubit? I can't find a way to modify a property of the cubit

[the remark this answers is not in the document]
Comment (ndmunoz10) [reply CarGurus App/lib/presentation/features/onboarding/onbording_screen.dart]: Sure, you can add the index in the OnboardingState class and then create a method to set the new index to the current index, eg: `void setPageIndex(final int newIndex) { emit(state.copyWith(currentPage: newIndex)) }`. Then you can just listen for state changes on the Widget and update the UI that you need to update

[the remark this answers is not in the document]
Comment (Giancarlo-Castaneda) [reply CarGurus App/lib/presentation/features/onboarding/onbording_screen.dart]: done, thank you

**Review by ndmunoz10 — APPROVED** [review]

**Pull request merged**